lib/temperature_monitor/temperature_handler.ex

```diff
@@ -0,0 +1,91 @@
+defmodule TemperatureMonitor.TemperatureHandler do
+  use GenServer
+
+  def start_link(stash_pid) do
+    {:ok, _pid} = GenServer.start_link(__MODULE__, stash_pid, name: __MODULE__)
+  end
+
+  def init(stash_pid) do
+    filelist = TemperatureMonitor.Stash.get_value stash_pid
+    monitored_paths = []
+     {:ok, {filelist, stash_pid, monitored_paths}}
+  end
+
+  def callback(:stop) do
+    IO.puts "STOP"
+  end
+
+  def callback(filepath, events) do
+    IO.inspect {filepath, events}
+  end
+
+  def monitor_path(base_path) do
+    GenServer.cast(__MODULE__, {:add_base_path, base_path})
+  end
+
+  def get_monitored_paths do
```

Comment:
BTW calling `Application.get_env` is really cheap because erlang manages all the configuration information in ETS tables so you could probably implement `TemperatureMonitor.Monitor.__dirs__/0` by just calling `Application.get_env` right there.